pkg/api/types.go

```diff
@@ -136,15 +153,24 @@ type InstanceSpec struct {
 }
 
 type InstanceStatus struct {
-	State InstanceState
+	Conditions []InstanceCondition
```

Comment:
I don't think there are any guarantees about conditions being present in the current API.

The k8s convention is not to prefer lists of named subobjects over maps, see kubernetes/kubernetes#2004 and https://github.com/kubernetes/kubernetes/blob/master/docs/devel/api-conventions.md#lists-of-named-subobjects-preferred-over-maps